src/action/common/configure_init_service.rs

```diff
@@ -75,6 +75,13 @@ impl ConfigureInitService {
             },
             #[cfg(target_os = "linux")]
             InitSystem::Systemd => {
+                // If /run/systemd/system exists, we can be reasonably sure the machine is booted
+                // with systemd: https://www.freedesktop.org/software/systemd/man/sd_booted.html
+                if !(Path::new("/run/systemd/system").exists() || which::which("systemctl").is_ok())
```

Comment:
This could be a case of a simple logic error, as in, the following would work:

```diff
-if !(Path::new("/run/systemd/system").exists() || which::which("systemctl").is_ok())
---
+if (!Path::new("/run/systemd/system").exists() || !which::which("systemctl").is_ok())
```

Although I'm not sure what the original intention was